app/models/friendship.rb

```diff
@@ -1,18 +1,20 @@
 class Friendship < ApplicationRecord
-  after_create :create_mutual_relationship
-  after_destroy :destroy_mutual_relationship
+  after_create :delete_request
 
-  belongs_to :user
-  belongs_to :friend, class_name: 'User'
+  belongs_to :sent_friend, class_name: 'User', foreign_key: 'adder_id'
+  belongs_to :received_friend, class_name: 'User', foreign_key: 'added_id'
+
+  validate :friend_is_self
 
   private
 
-  def create_mutual_relationship
-    Friendship.find_or_create_by(user_id: friend.id, friend_id: user.id)
+  def friend_is_self
+    errors.add(:adder, "cannot add self as friend") if adder_id == added_id
   end
 
-  def destroy_mutual_relationship
-    friendship = Friendship.find_by(user_id: friend.id, friend_id: user.id)
-    friendship&.destroy
+  def delete_request
```

Comment:
Shouldn't this method be in `friend_request.rb` instead?

Reply:
I thought the request was suppose to be deleted after a friendship object is created and that's why I put it in the friendship model.

Reply:
@kvnlay! Haha i am very sorry. Yeah, my mistake. You are right this is where it's supposed to be. I thought I was looking at another method.